crates/wasi-common/yanix/src/fcntl.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{
+    file::{FdFlag, OFlag},
+    Errno, Result,
+};
+use std::os::unix::prelude::*;
+
+pub unsafe fn dup_fd(fd: RawFd, close_on_exec: bool) -> Result<RawFd> {
+    // Both fcntl commands expect a RawFd arg which will specify
+    // the minimum duplicated RawFd number. In our case, I don't
+    // think we have to worry about this that much, so passing in
+    // the RawFd descriptor we want duplicated
+    Errno::from_result(if close_on_exec {
+        libc::fcntl(fd, libc::F_DUPFD_CLOEXEC, fd)
+    } else {
+        libc::fcntl(fd, libc::F_DUPFD, fd)
+    })
+}
+
+pub fn get_fd_flags(fd: RawFd) -> Result<FdFlag> {
```

Comment:
This (and the similar functions below) should also be unsafe, due to taking raw file descriptors.

Reply:
Ahh, of course, thanks for catching that! :-)

Reply:
Done!